test/e2e/azureidentity/azureidentity.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	deployFilePath := path.Join(templateOutputPath, name+".yaml")
+	deployFile, err := os.Create(deployFilePath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create a deployment file from aadpodidentity.yaml")
+	}
+	defer deployFile.Close()
+
+	deployData := struct {
+		SubscriptionID string
+		ResourceGroup  string
+		ClientID       string
+		Name           string
+	}{
+		subscriptionID,
+		resourceGroup,
+		clientID,
+		name,
+	}
+	if err := t.Execute(deployFile, deployData); err != nil {
+		return errors.Wrap(err, "Failed to create a deployment file from aadpodidentity.yaml")
+	}
+
+	cmd := exec.Command("kubectl", "apply", "-f", deployFilePath)
+	util.PrintCommand(cmd)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to deploy AzureIdentity to the Kubernetes cluster")
+	}
+
+	return nil
+}
+
+// DeleteOnCluster will delete an Azure Identity on a Kubernetes cluster
+func DeleteOnCluster(name, templateOutputPath string) error {
+	cmd := exec.Command("kubectl", "delete", "-f", path.Join(templateOutputPath, name+".yaml"), "--ignore-not-found")
+	util.PrintCommand(cmd)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to delete AzureIdentity from the Kubernetes cluster")
+	}
+
+	return nil
+}
+
+// GetClientID will return the client id of a user-assigned identity on Azure
+func GetClientID(resourceGroup, name string) (string, error) {
+	cmd := exec.Command("az", "identity", "show", "-g", resourceGroup, "-n", name, "--query", "clientId", "-otsv")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get the clientID from the identity in Azure")
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+// GetPrincipalID will return the principal id (objecet id) of a user-assigned identity on Azure
+func GetPrincipalID(resourceGroup, name string) (string, error) {
+	cmd := exec.Command("az", "identity", "show", "-g", resourceGroup, "-n", name, "--query", "principalId", "-otsv")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get the principalID from the identity in Azure")
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+// GetAll will return a list of AzureIdentity deployed on a Kubernetes cluster
+func GetAll() (*aadpodid.AzureIdentityList, error) {
+	cmd := exec.Command("kubectl", "get", "AzureIdentity", "-ojson")
+	util.PrintCommand(cmd)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get AzureIdentity from the Kubernetes cluster")
+	}
+
+	list := aadpodid.AzureIdentityList{}
+	err = json.Unmarshal(out, &list)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to unmarshall json")
+	}
+
+	return &list, nil
+}
+
+// WaitOnReaderRoleAssignment will block until the assignement of 'Reader' role to an identity is executed successfully
+func WaitOnReaderRoleAssignment(resourceGroup, name string) (bool, error) {
```

Comment:
Aren't these common methods which does not require any test specific information. Can you move such methods to a common part of the code, please ?

Reply:
Moved all the non test code to `test/common`